### PR TITLE
Add settings var to choose application metrics to collect

### PR DESCRIPTION
**Pull request by Ferril**
# What this PR does
Adds settings var `METRICS_TO_COLLECT` to choose what metrics should be collected by `ApplicationMetricsCollector`.
It allows to collect different application metrics using different exporters.

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

**Review by mderynck — APPROVED** [review]

File: engine/apps/metrics_exporter/metrics_collectors.py [hunk 1]
```diff
@@ -61,29 +69,33 @@ def __init__(self):
...
         self.recalculate_cache_for_missing_org_ids(org_ids, missing_org_ids)
 
-        yield alert_groups_total
-        yield alert_groups_response_time_seconds
-        yield user_was_notified
+        for metric in metrics:
+            yield metric
 
-    def _get_alert_groups_total_metric(self, org_ids):
+    def _get_alert_groups_total_metric(self, org_ids: set) -> typing.Tuple[Metric, set]:
```
Comment (joeyorlando) [inline engine/apps/metrics_exporter/metrics_collectors.py:98]: _nit_
```suggestion
    def _get_alert_groups_total_metric(self, org_ids: set[int]) -> typing.Tuple[Metric, set[int]]:
```

File: engine/apps/metrics_exporter/metrics_collectors.py [hunk 2]
```diff
@@ -160,34 +173,48 @@ def _get_response_time_metric(self, org_ids):
...
+        # Labels values should have the same order as _integration_labels_with_state
+        labels_values = [
+            integration_data["integration_name"],  # integration
+            integration_data["team_name"],  # team
+            integration_data["org_id"],  # grafana org_id
+            integration_data["slug"],  # grafana instance slug
+            integration_data["id"],  # grafana instance id
+        ]
+        labels_values = list(map(str, labels_values))
+        return labels_values
```
Comment (joeyorlando) [inline engine/apps/metrics_exporter/metrics_collectors.py:199]: _nit_
```suggestion
        return list(map(str, labels_values))
```

File: engine/apps/metrics_exporter/metrics_collectors.py [hunk 3]
```diff
@@ -160,34 +173,48 @@ def _get_response_time_metric(self, org_ids):
...
+    def _get_labels_from_user_data(self, user_data: UserWasNotifiedOfAlertGroupsMetricsDict) -> typing.List[str]:
+        # Labels values should have the same order as _user_labels
+        labels_values = [
+            user_data["user_username"],  # username
+            user_data["org_id"],  # grafana org_id
+            user_data["slug"],  # grafana instance slug
+            user_data["id"],  # grafana instance id
+        ]
+        labels_values = list(map(str, labels_values))
+        return labels_values
```
Comment (joeyorlando) [inline engine/apps/metrics_exporter/metrics_collectors.py:210]: _nit_
```suggestion
        return list(map(str, labels_values))
```

File: engine/apps/metrics_exporter/metrics_collectors.py [hunk 4]
```diff
@@ -118,7 +122,25 @@ def _get_alert_groups_total_metric(self, org_ids):
         missing_org_ids = org_ids - processed_org_ids
         return alert_groups_total, missing_org_ids
 
-    def _get_response_time_metric(self, org_ids):
+    def _get_user_was_notified_of_alert_groups_metric(self, org_ids: set) -> typing.Tuple[Metric, set]:
```
Comment (joeyorlando) [inline engine/apps/metrics_exporter/metrics_collectors.py:125]: _nit_
```suggestion
    def _get_user_was_notified_of_alert_groups_metric(self, org_ids: set[int]) -> typing.Tuple[Metric, set[int]]:
```

File: engine/apps/metrics_exporter/metrics_collectors.py [hunk 5]
```diff
@@ -118,7 +122,25 @@ def _get_alert_groups_total_metric(self, org_ids):
...
+        for org_key, users in org_users.items():
+            for _, user_data in users.items():
+                labels_values: typing.List[str] = self._get_labels_from_user_data(user_data)
+                user_was_notified.add_metric(labels_values, user_data["counter"])
+            org_id_from_key = RE_USER_WAS_NOTIFIED_OF_ALERT_GROUPS.match(org_key).groups()[0]
+            processed_org_ids.add(int(org_id_from_key))
+        missing_org_ids = org_ids - processed_org_ids
+        return user_was_notified, missing_org_ids
+
+    def _get_response_time_metric(self, org_ids: set) -> typing.Tuple[Metric, set]:
```
Comment (joeyorlando) [inline engine/apps/metrics_exporter/metrics_collectors.py:143]: _nit_
```suggestion
    def _get_response_time_metric(self, org_ids: set[int]) -> typing.Tuple[Metric, set[int]]:
```

File: engine/apps/metrics_exporter/metrics_collectors.py [hunk 6]
```diff
@@ -197,7 +224,7 @@ def _update_new_metric(self, metric_name, org_ids, missing_org_ids):
                 start_recalculation_for_new_metric.apply_async((metric_name,))
         return missing_org_ids
 
-    def recalculate_cache_for_missing_org_ids(self, org_ids, missing_org_ids):
+    def recalculate_cache_for_missing_org_ids(self, org_ids: set, missing_org_ids: set) -> None:
```
Comment (joeyorlando) [inline engine/apps/metrics_exporter/metrics_collectors.py:227]: _nit_
```suggestion
    def recalculate_cache_for_missing_org_ids(self, org_ids: set[int], missing_org_ids: set[int]) -> None:
```

**Review by joeyorlando — APPROVED** [review]
_super_ minor nits, LGTM 👍

**Pull request merged**